CONTRIBUTING.md

```diff
@@ -543,10 +533,47 @@ Refer datalad/config.py for information on how to add these environment variable
 - *DATALAD_ALLOW_FAIL*:
   Instructs `@never_fail` decorator to allow to fail, e.g. to ease debugging.
 
-# Changelog section
+# Release(s) workflow
+
+## Branches
+
+- `master`: changes toward the next `MAJOR.MINOR.0` release (e.g., `0.13` as of time of writing).
+  It is also used for release candidates (with `rcX` suffix)
+- `maint`: bug fixes and minor enhancements for the latest released `MAJOR.MINOR.PATCH`
+ (e.g., `0.12.7` as of time of writing)
+- `MAJOR.MINOR`: generally not used, unless some bugfix release with a crticial bugfix is needed.
+
+## Workflow
```

Comment:
I think it's fine to have this section, but I think more importantly we should resurrect and tweak the text we had in .github/PULL_REQUEST_TEMPLATE.md before 82e9d6a57 (MNT: Drop "Choose a base" item from PR template checklist, 2020-01-16).

Reply:
This is for `Release Management` section, so not sure how PR template relates to that.